com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java

```diff
@@ -902,4 +902,23 @@ public static boolean isArchiveExcluded(Project project) {
         return project.option("exclude-archive", "false").equals("false");
     }
 
+
+    // must have two or more segments
+    // each segment must start with a letter
+    // all characters must be alphanumeric or an underscore
+    public static boolean isValidAndroidPackageName(String packageName) {
+        return packageName.matches("^[a-zA-Z][a-zA-Z0-9_]*(\\.[a-zA-Z][a-zA-Z0-9_]*)+$");
+    }
+
+
+    // must have two or more segments
+    // each segment must start with a letter
+    // all characters must be alphanumeric, an underscore or a hyphen
+    public static boolean isValidAppleBundleIdentifier(String bundleIdentifier) {
+        return bundleIdentifier.matches("^[a-zA-Z][a-zA-Z0-9_-]*(\\.[a-zA-Z][a-zA-Z0-9_-]*)+$");
+    }
```

Comment:
I see these functions are isolated. So we probably will be able to reuse them from the editor in the future.

Reply:
Yes, we should probably look into field validation for these in the editor. I'll take a quick look before merging this.

Reply:
We currently have no support for field validation in game.project. This fix will have to be for bob only for now.